wordpress-mobile/WordPress-Android
# Quick Start for Existing Users V2 - Add Metrics

Closes #16353

This PR updates tracking for all quick start events to add `site_type` based on quick start type values: `existing_site` or `new_site`.

Corresponding iOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/18560

To test:
1. Make sure that (todo) trackings are added for the newly added `get_to_know_app` quick start type in be833a2.
2. Make sure that [all quick start events](https://github.com/wordpress-mobile/WordPress-Android/blob/fc72b41dc20fc9ae14bcc38b4d6c906eefa02d9d/libs/analytics/WordPressAnalytics/src/main/java/org/wordpress/android/analytics/AnalyticsTracker.java#L626-L689) are tracked using newly added `QuickStartTracker`.

Event | Commit Message | Commit
-- | -- | --
QUICK_START_STARTED | Track quick start started using quick start tracker | f58bb5ca
QUICK_START_TAPPED | Track quick start card item clicked (from menu tab) using quick start tracker, Track quick start card item clicked (from dashboard tab) using quick start tracker | 09817d4a, 0dab8bd
QUICK_START_TASK_DIALOG_VIEWED | Track quick start all task completed, qs task dialog viewed, qs task dialog buttons tapped using quick start tracker | d53986de
QUICK_START_TASK_DIALOG_NEGATIVE_TAPPED | -do- | -do-
QUICK_START_TASK_DIALOG_POSITIVE_TAPPED | Track quick start remove dialog buttons tapped, qs request dialog viewed, qs request dialog buttons tapped using quick start tracker | 49ff0b94
QUICK_START_REMOVE_DIALOG_NEGATIVE_TAPPED | -do- | -do-
QUICK_START_REMOVE_DIALOG_POSITIVE_TAPPED | -do- | -do-
QUICK_START_TYPE_CUSTOMIZE_VIEWED | Track quick start type viewed, dismissed, quick start list expanded, collapsed using quick start tracker | 242fabea
QUICK_START_TYPE_GROW_VIEWED | -do- | -do-
QUICK_START_TYPE_GET_TO_KNOW_APP_VIEWED | -do- | -do-
QUICK_START_TYPE_CUSTOMIZE_DISMISSED | -do- | -do-
QUICK_START_TYPE_GROW_DISMISSED | -do- | -do-
QUICK_START_TYPE_GET_TO_KNOW_APP_DISMISSED | -do- | -do-
QUICK_START_LIST_CUSTOMIZE_COLLAPSED | -do- | -do-
QUICK_START_LIST_GROW_COLLAPSED | -do- | -do-
QUICK_START_GET_TO_KNOW_APP_COLLAPSED | -do- | -do-
QUICK_START_LIST_CUSTOMIZE_EXPANDED | -do- | -do-
QUICK_START_LIST_GROW_EXPANDED | -do- | -do-
QUICK_START_GET_TO_KNOW_APP_EXPANDED | -do- | -do-
QUICK_START_LIST_CREATE_SITE_SKIPPED | Track quick start list task tapped, skipped using quick start tracker | 3e5c16d8
QUICK_START_LIST_UPDATE_SITE_TITLE_SKIPPED | -do- | -do-
QUICK_START_LIST_VIEW_SITE_SKIPPED | -do- | -do-
QUICK_START_LIST_ADD_SOCIAL_SKIPPED | -do- | -do-
QUICK_START_LIST_PUBLISH_POST_SKIPPED | -do- | -do-
QUICK_START_LIST_FOLLOW_SITE_SKIPPED | -do- | -do-
QUICK_START_LIST_UPLOAD_ICON_SKIPPED | -do- | -do-
QUICK_START_LIST_CHECK_STATS_SKIPPED | -do- | -do-
QUICK_START_LIST_EXPLORE_PLANS_SKIPPED | -do- | -do-
QUICK_START_LIST_EDIT_HOMEPAGE_SKIPPED | -do- | -do-
QUICK_START_LIST_REVIEW_PAGES_SKIPPED | -do- | -do-
QUICK_START_LIST_CHECK_NOTIFICATIONS_SKIPPED | -do- | -do-
QUICK_START_LIST_CREATE_SITE_TAPPED | -do- | -do-
QUICK_START_LIST_UPDATE_SITE_TITLE_TAPPED | -do- | -do-
QUICK_START_LIST_VIEW_SITE_TAPPED | -do- | -do-
QUICK_START_LIST_ADD_SOCIAL_TAPPED | -do- | -do-
QUICK_START_LIST_PUBLISH_POST_TAPPED | -do- | -do-
QUICK_START_LIST_FOLLOW_SITE_TAPPED | -do- | -do-
QUICK_START_LIST_UPLOAD_ICON_TAPPED | -do- | -do-
QUICK_START_LIST_CHECK_STATS_TAPPED | -do- | -do-
QUICK_START_LIST_EXPLORE_PLANS_TAPPED | -do- | -do-
QUICK_START_LIST_EDIT_HOMEPAGE_TAPPED | -do- | -do-
QUICK_START_LIST_REVIEW_PAGES_TAPPED | -do- | -do-
QUICK_START_LIST_CHECK_NOTIFICATIONS_TAPPED | -do- | -do-
QUICK_START_CREATE_SITE_TASK_COMPLETED | Track quick start completed using quick start tracker | 12507cc4
QUICK_START_UPDATE_SITE_TITLE_COMPLETED | -do- | -do-
QUICK_START_VIEW_SITE_TASK_COMPLETED | -do- | -do-
QUICK_START_SHARE_SITE_TASK_COMPLETED | -do- | -do-
QUICK_START_PUBLISH_POST_TASK_COMPLETED | -do- | -do-
QUICK_START_FOLLOW_SITE_TASK_COMPLETED | -do- | -do-
QUICK_START_UPLOAD_ICON_COMPLETED | -do- | -do-
QUICK_START_CHECK_STATS_COMPLETED | -do- | -do-
QUICK_START_EXPLORE_PLANS_COMPLETED | -do- | -do-
QUICK_START_EDIT_HOMEPAGE_TASK_COMPLETED | -do- | -do-
QUICK_START_REVIEW_PAGES_TASK_COMPLETED | -do- | -do-
QUICK_START_CHECK_NOTIFICATIONS_TASK_COMPLETED | -do- | -do-
QUICK_START_ALL_TASKS_COMPLETED | Track quick start all task completed, qs task dialog viewed, qs task dialog buttons tapped using quick start tracker | d53986de
QUICK_START_REQUEST_VIEWED | Track quick start remove dialog buttons tapped, qs request dialog viewed, qs request dialog buttons tapped using quick start tracker | 49ff0b94
QUICK_START_REQUEST_DIALOG_NEGATIVE_TAPPED | -do- |  
QUICK_START_REQUEST_DIALOG_POSITIVE_TAPPED | -do- | 49ff0b94
QUICK_START_NOTIFICATION_DISMISSED | Track quick start notification dimissed using quick start tracker | 98658e4f
QUICK_START_NOTIFICATION_SENT | Track quick start notification sent and tapped using quick start tracker | 6feb9f8b
QUICK_START_NOTIFICATION_TAPPED | -do- | -do-
QUICK_START_HIDE_CARD_TAPPED | Track quick start hide card tapped, quick start remove card tapped using quick start tracker | ec485b0e
QUICK_START_REMOVE_CARD_TAPPED | -do- | -do-

3. Enable tracking using `Me` -> `App Settings` -> `Privacy settings` -> `Collect information`.
4. Launch the app and interact with the quick start card so that a few quick start events are logged.
5. Make sure that the logs contain `site_type` key with an appropriate quick start type value: `existing_site` or `new_site`.
6. Make sure that the logs contain all the existing properties for the `Stat` (like default tab experiment properties).


Merge Instructions:

1. Make sure #16541 PR is merged to trunk.
2. Merge trunk to this PR.
3. Make sure that tracking for the upload media quick start task are in order.
4. Remove `Not Ready for Merge` label.
5. Merge the PR.


## Regression Notes
1. Potential unintended areas of impact
N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
